integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -444,6 +450,18 @@ def test_hash_groupby_collect_set(data_gen):
             .groupby('a')
             .agg(f.sort_array(f.collect_set('b')), f.count('b')))
 
+@approximate_float
+@ignore_order(local=True)
+@incompat
+@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op, ids=idfn)
+@pytest.mark.xfail(reason="the result order from collect-set can not be ensured for CPU and GPU."
```

Comment:
Is there a follow on issue for this in particular?  I personally would prefer a skip over an xfail, or is there a way that we can do a simple small test where there will only ever be a single value in the set so we can at least test that case.

Reply:
Done, the new commit has added a new test case with unique group by key for this.

Reply:
@revans2 Could you help to review it again? Thx